kernel/arch/dreamcast/hardware/spu.c

```diff
@@ -226,17 +256,19 @@ void spu_memset_sq(uintptr_t dst, uint32_t what, size_t length) {
 void spu_reset_chans(void) {
     int i;
     g2_fifo_wait();
-    g2_write_32(SNDREGADDR(0x2800), 0);
+    g2_ctx_t ctx = g2_lock();
```

Comment:
Cosmetic change, but could you declare the `g2_ctx_t ctx;` at the beginning of the function? (can still be locked here)